neuralpp/symbolic/z3_expression.py

```diff
@@ -377,31 +383,39 @@ def number_of_arguments(self) -> int:
 z3_false = z3.BoolVal(False)
 
 
-def _extract_key_value_from_assertions_helper(assertions: List[z3.ExprRef] | z3.AstVector, dict_: Dict[str, Any]):
-    def is_value(v):
-        return z3.is_int_value(v) or z3.is_rational_value(v) or z3.is_algebraic_value(v) or z3.is_fp_value(v)
-
-    def is_key(k):
-        return z3.is_const(k) and not is_value(k)
-
+def _traverse_equalities(assertions: List[z3.ExprRef], accumulator: Callable[[z3.ExprRef, z3.ExprRef], type(None)]):
```

Comment:
Rewrote this function to make it general-purpose, with an `accumulator`, so other traversing functions can use this.